ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -505,17 +506,19 @@ PartitionId getRandomWritablePartition(String partitionClass, List<PartitionId>
     }
 
     /**
-     * Check whether the parition has has enough replicas up for write operations to try. Enough replicas is considered
-     * to be all local replicas if such information is available. In case localDatacenterName is not available, all of
-     * the partition's replicas should be up.
+     * Check whether the partition has enough eligible replicas for write operations to try. Here, "eligible" means
+     * replica is up and in required states for PUT request. Enough replicas is considered to be all local replicas if
+     * such information is available. In case localDatacenterName is not available, all of the partition's replicas
+     * should be up.
      * @param partitionId the {@link PartitionId} to check.
-     * @return true if enough replicas are up; false otherwise.
+     * @return true if enough replicas are eligible; false otherwise.
      */
-    private boolean areEnoughReplicasForPartitionUp(PartitionId partitionId) {
+    private boolean hasEnoughEligibleReplicasForPut(PartitionId partitionId) {
```

Comment:
This is a nitpick. Feel free to ignore, if you don't feel like it.

I would suggest that either we name both methods (`getRandomWritablePartition`, `hasEnoughEligibleReplicasForPut`) to one of these names
1.  (`getRandomWritablePartition`, `hasEnoughEligibleWritableReplicas`)
2.  (`getRandomPartitionForPut`, `hasEnoughEligibleReplicasForPut`)

But if we decide to take approach 2, then we should also rename these:
`ClusterMap::getRandomWritablePartition` to `ClusterMap::getRandomPartitionForPut`
`PartitionLayout::getRandomWritablePartition` to `PartitionLayout::getRandomPartitionForPut`

Reply:
Option 1 sounds like a good name to me without requiring many public interface changes

Reply:
sure, let's take option 1